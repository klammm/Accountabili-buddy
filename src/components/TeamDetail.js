import React from 'react';
import { StyleSheet } from 'react-native';
import { CardSection, Card } from './common';


// REMINDER:
// Button when pressed needs to join a team
const TeamDetail = ({ team }) => {
  const { name, team_image_url } = team;
  const { thumbnailContainerStyle, thumbnailStyle } = styles;

  return (
    <Card>
      <CardSection>
        <View style={thumbnailContainerStyle}>
          <Image style={thumbnailStyle} source={{ uri: team_image_url }} />
        </View>
      </CardSection>

      <CardSection>
        <Button>
          { name }
        </Button>
      </CardSection>
    </Card>
  );
};

const styles = StyleSheet.create({
  thumbnailStyle: {
    height: 50,
    width: 50
  },
  thumbnailContainerStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    marginRight: 10
  }
})

export default TeamDetail
