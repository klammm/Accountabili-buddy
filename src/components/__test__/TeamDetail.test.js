import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TeamDetail from '../TeamDetail';


describe('tests the TeamDetail component', () => {
  it('renders TeamDetail component', () => {
  const component = shallow(
    <TeamDetail
      team={{}}
    />
  );
    expect(toJson(component)).toMatchSnapshot();
  });
});
