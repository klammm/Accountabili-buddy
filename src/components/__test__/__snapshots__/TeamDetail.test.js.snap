// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the TeamDetail component renders TeamDetail component 1`] = `
<Card>
  <CardSection>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </CardSection>
  <CardSection>
    <Button />
  </CardSection>
</Card>
`;
